package/gluon-status-page-api/src/stations.c

```diff
@@ -58,8 +61,44 @@ bool interface_is_valid(const char *ifname) {
   return access(path, F_OK) == 0;
 }
 
+int interface_get_lowest(const char *ifname, char *hwifname);
+int interface_get_lowest(const char *ifname, char *hwifname) {
+  glob_t globbuf;
+  char *fnamebuf = alloca(1 + strlen(VIRTIF_PREFIX) + IF_NAMESIZE +
+                          strlen(LOWERGLOB_SUFFIX));
+  char *lowentry = NULL;
+
+  sprintf(fnamebuf, "%s%s%s", VIRTIF_PREFIX, ifname, LOWERGLOB_SUFFIX);
+  glob(fnamebuf, GLOB_NOSORT, NULL, &globbuf);
+
+  if (globbuf.gl_pathc == 1)
+    lowentry = strdup(globbuf.gl_pathv[0]);
+
+  globfree(&globbuf);
+
+  if (!lowentry) {
+    strncpy(hwifname, ifname, IF_NAMESIZE);
```

Comment:
This should be `IFNAMESIZE-1`. While I don't see any way how the name could ever be too long (so it isn't really a problem in practice), the maximum ifname length is `IFNAMESIZE-1`; this will ensure that hwifname will always be zero-terminated.